### PR TITLE
Decentralized Activation

### PR DESCRIPTION
**Pull request by FlorianRappl**
Refactored to use `decentralized-activation`.

**Important**:
The module name has to be renamed with a suffix of `-app`. As outlined last week we should also rename the repository then.

### CODE REVIEW

**Comment by HerbertYiga** [comment]
hi @FlorianRappl   thanks for the works on this,i just got a couple your pull requests with the same description,can these pull requests be in a single pr?

**Comment by FlorianRappl** [comment]
> can these pull requests be in a single pr?

They are all in different repositories, so I think, unfortunately, no.

File: src/index.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  "../translations",
+  false,
+  /.json$/,
+  "lazy"
+);
+
+function setupOpenMRS() {
+  return {
+    lifecycle: () => import("./openmrs-esm-login"),
+    activate: (location) => routePrefix("login", location),
```
Comment (brandones) [inline src/index.ts:15]:
```suggestion
    activate: "login"
```

File: src/login/login.component.tsx [hunk 2]
```diff
@@ -13,7 +13,7 @@ interface LoginReferrer {
 }
 
 interface LoginProps
-  extends RouteComponentProps<{}, StaticContext, LoginReferrer> {}
+  extends RouteComponentProps<{}, undefined, LoginReferrer> {}
```
Comment (brandones) [inline src/login/login.component.tsx:16]: What's this about?

Comment (brandones) [reply src/login/login.component.tsx:16]: ?

Comment (FlorianRappl) [reply src/login/login.component.tsx:16]: This type (`StaticContext`) does not matter - unfortunately its not re-exported from `react-router-dom`, but I don't want (even for types) mention `react-router`.

**Pull request merged**